lua/pac3/core/client/bones.lua

```diff
@@ -366,6 +406,31 @@ do -- bone manipulation for boneanimlib
 	local ManipulateBoneAngles = entmeta.ManipulateBoneAngles
 	local ManipulateBoneJiggle = entmeta.ManipulateBoneJiggle
 
+	function pac.SetSpecialBoneMatrix(ent, boneID, mMatrix)
+		ent.pac_special_bone_anims = ent.pac_special_bone_anims or {}
+		ent.pac_special_bone_anims[boneID] = mMatrix
+	end
+
+	function pac.ClearSpecialBones(ent)
+		if(ent.pac_special_bone_anims)then
+			for boneID, _ in pairs(ent.pac_special_bone_anims) do
+				ent.pac_special_bone_anims[boneID] = nil
+			end
+		end
+	end
+
+	function pac.ManipulateSpecialBone(ent, boneID, pos, ang)
+		if(ent.pac_special_bone_anims and ent.pac_special_bone_anims[boneID])then
+			local matrix = ent.pac_special_bone_anims[boneID]
+			pos = pos + matrix:GetTranslation()
+			ang = ang + matrix:GetAngles()
```

Comment:
This should be somethink like:
```lua
result = matrix1 * matrix2
pos = result:GetTranslation()
ang = result:GetAngles()
```
Adding euler angles is almost never good

Reply:
I didn't know that, thanks! Is it because of gimbal lock?

I've just pushed a commit with what I think is a proper way of implementing what you described.

Reply:
Looks good. Yeah gimble lock and also because adding euler angle values won't accurately represent a rotation plus another rotation. It only does if only one of the components of each are nonzero.